x-pack/legacy/plugins/maps/public/layers/styles/vector/vector_style.js

```diff
@@ -484,15 +453,12 @@ export class VectorStyle extends AbstractStyle {
           feature.properties[computedName] = styleValue;
         }
       }
-      tmpFeatureIdentifier.source = sourceId;
+      tmpFeatureIdentifier.source = mbSourceId;
       tmpFeatureIdentifier.id = feature.id;
       mbMap.setFeatureState(tmpFeatureIdentifier, tmpFeatureState);
     }
 
-    const hasGeoJsonProperties = styleFields.some(({ supportsFeatureState }) => {
-      return !supportsFeatureState;
-    });
-    return hasGeoJsonProperties;
+    return featureStateParams.some(({ supportsFeatureState }) => !supportsFeatureState);
```

Comment:
Since you are removing the variable name that sort of explains what `some` is looking for, could you add a comment either here or in the function header about what this return value is. Something like "boolean indicating if styles do not support feature-state and values are stored in geojson properties."